File: movie_view.py

import numpy as np
import cv2
import argparse


def main():

    parser = argparse.ArgumentParser()
    parser.add_argument('--src')
    args = parser.parse_args()

    cap = cv2.VideoCapture(args.src)

    name = "effect"
    fps = cap.get(cv2.CAP_PROP_FPS)

    while cap.isOpened():
        _ret, frame = cap.read()

        k = cv2.waitKey(int(fps)) & 0xFF
        if k == 27:
            break
        
        if frame is None:
            break

        ## [TODO] image process ##
        dst = frame
        cv2.imshow(name,dst)

    cap.release()


if __name__ == '__main__':
    main()

File: movie_view.cpp
#include <iostream>
#include <opencv2/core.hpp>
#include <opencv2/core/utility.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/videoio.hpp>
#include <opencv2/highgui.hpp>

int main(int argc, const char* argv[])
{
    cv::String keys = "{src||}""{cascade||}";
    cv::CommandLineParser parser(argc, argv, keys);
    cv::String src_path = parser.get<cv::String>("src");

    cv::VideoCapture capture(src_path);
    if (!capture.isOpened()){
        //error in opening the video input
        std::cerr << "Unable to open file!" << std::endl;
        return 0;
    }

    std::string name = "Effect";
    cv::namedWindow(name, cv::WINDOW_AUTOSIZE);
    int fps = int(capture.get(cv::CAP_PROP_FPS));

    while(true){
        cv::Mat frame;
        capture >> frame;

        if (frame.empty())
            break;

        cv::imshow(name, frame);
        frame.release();
        
        int key = cv::waitKey(fps);
        if (key == 27)
            break;

    }

    capture.release();
    cv::destroyAllWindows();
    
    return 0;
}
